fastify.js

```diff
@@ -313,21 +313,15 @@ function build (options) {
     return instance
   }
 
-  function RoutePrefix () {
-    this.prefix = ''
-  }
+  function buildRoutePrefix (instancePrefix, pluginPrefix) {
+    if (!pluginPrefix) {
+      return instancePrefix
+    }
 
-  function buildRoutePrefix (r, opts) {
-    const _RoutePrefix = Object.create(opts)
-    const R = _RoutePrefix
-    R.prefix = r.prefix
-    if (typeof opts.prefix === 'string') {
-      if (opts.prefix[0] !== '/') {
-        opts.prefix = '/' + opts.prefix
-      }
-      R.prefix += opts.prefix
+    if (pluginPrefix[0] !== '/') {
+      pluginPrefix = '/' + pluginPrefix
     }
-    return R
+    return instancePrefix + pluginPrefix
```

Comment:
What happen with `buildRoutePrefix('/foo/', 'bar)`? the returned string is `/foo//bar`

Reply:
Yes. I think that's what the discussion in #584 is about.